src/cpp/dpm_sampler.cpp

```diff
@@ -1803,6 +1832,8 @@ void cleanup()
     free(c_ptr);
     free(pbase);
     free(cbase);
+    free(res_dist);
+    free(res);
     free(log_pbase);
     free(h);
```

Comment:
god I can't wait till all this manual memory management is gone

Reply:
I have a commit in that direction on my  TODO list.
(well, had one, until the whole hand-accident thing.
You'll need to be a little patient here...)